sound/soc/soc-core.c

```diff
@@ -2780,12 +2780,12 @@ int snd_soc_of_parse_audio_routing(struct snd_soc_card *card,
 			propname);
 		return -EINVAL;
 	}
-	num_routes /= 2;
 	if (!num_routes) {
 		dev_err(card->dev, "ASoC: Property '%s's length is zero\n",
 			propname);
 		return -EINVAL;
 	}
+	num_routes /= 2;
```

Comment:
this one is actually strange... Isn't it a false positive? Assuming cppcheck actually only checks preprocessing. In fact if you look at `of_property_count_strings()` and then at `of_property_read_string_helper()` then indeed 0 is never returned there! For 0 `-ENODATA` is returned. But I wouldn't expect cppcheck to check that? But anyway wouldn't the best fix be to actually drop the `(!num_routes)` check?

Reply:
< 0 is an error
odd values are not possible, so 1 is not allowed

at this point it's either 0, or 2+

What cppcheck detected is that if num_routes = 2+, then after the division by two num_routes cannot be zero.

if zero cannot happen, then we can remove this condition, but that's not something cppcheck would verify on its own.

Reply:
cppcheck complained that "Condition '!num_routes' is always false" and that's exactly what surprised me - that seems to suggest, that it recognised, that `num_routes` cannot be 0, for which it had to check the `of_property_count_strings()` function which would be a strange thing to do for a static code analyser...

Reply:
I removed this test as suggested, it cannot happen. Thanks for the feedback and detailed reviews, much appreciated @lyakh